roles/openshift_control_plane/tasks/main.yml

```diff
@@ -99,7 +99,7 @@
 
 - name: Create the request header ca file if needed
   copy:
-    dest: "{{ item.clientCA if 'clientCA' in item and '/' in item.clientCA else '/etc/origin/master/' ~ item.clientCA | default('request_header_ca.crt') }}"
+    dest: "/etc/origin/master/request_header_ca.crt"
```

Comment:
We need to evaluate the boolean cases in this PR as well.

It seems that this file is only created of the clientCA key is defined in an item in openshift_master_identity_providers.  Since we've updated the docs, that field is no longer there, thus this field might not get defined.

Probably just need to force people to migrate these CA keys to not include a leading '/' char, and then just put the file path in /etc/origin/master/* wherever they specify.

Reply:
>Probably just need to force people to migrate these CA keys to not include a leading '/' char, and then just put the file path in /etc/origin/master/* wherever they specify

Not sure what would be the usecase for this, although its potentially dangerous, as user may rewrite etcd/api certs if the name is `ca-bundle.crt` or `ca.crt` or `admin.crt` , which are pretty common.

Reply:
I don't care where we put the files or what we name them, as long as they are in /etc/origin/master they will be fine.  We already have logic that will block existing users from upgrades if any paths outside of /etc/origin/master are found in master's config file.

We need to implement similar logic for the identity provider dictionary for new installs, backport to 3.10.  The change as-is here would break some logic, and define variables when they might otherwise be undefined by the user.

For example, https://github.com/openshift/openshift-ansible/pull/9731/files#diff-54774cd1a74c81a803bc9a139d03f9d6R109 will never be true because we're now telling users not to define the 'clientCA' key in the example.

Reply:
Seems like we should just get rid of that conditional across these set of tasks?

Reply:
>We already have logic that will block existing users from upgrades if any paths outside of /etc/origin/master are found in master's config file

Identity providers are skipped there, as this part of the code is expected to do that. Also, this check won't forbid rewriting existing certs.

> .. will never be true because we're now telling users not to define the 'clientCA' key in the example.

Correct, removed this clause.